### PR TITLE
Menu IO

### PR DESCRIPTION
**Pull request by OmniBlade**
Restores keyboard and mouse handler and implements missing functions for drawing the mouse cursor. Follows #12.

### CODE REVIEW

File: redalert/win32lib/mouseww.cpp [hunk 1]
```diff
@@ -189,13 +191,13 @@ void WWMouseClass::Process_Mouse(void)
...
     // threads that may try and lock the screen buffer
     //
-    // Block_Mouse(Screen->Get_Graphic_Buffer());
+    Block_Mouse(Screen->Get_Graphic_Buffer());
 
     //
     // If the screen is already locked by another thread then just exit
     //
     if (Screen->Get_LockCount() != 0) {
-        // Unblock_Mouse(Screen->Get_Graphic_Buffer());
```
Comment (hifi) [inline redalert/win32lib/mouseww.cpp]: Was this supposed to be uncommented?

File: common/wwmouse.c [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+                ++dst;
+            }
+
+            src += src_pitch;
+            dst += dst_pitch;
+        }
+    }
+}
+
+void* __cdecl ASM_Set_Mouse_Cursor(void* thisptr, int hotspotx, int hotspoty, void* cursor)
```
Comment (hifi) [inline common/wwmouse.c:187]: What's with the `ASM_` prefix?

File: redalert/win32lib/mouseww.cpp (hunk 1 above)
Comment (OmniBlade) [reply redalert/win32lib/mouseww.cpp]: Looking at how the original works I don't think so.

**Comment by hifi** [comment]
When testing this with MinGW branch on Wine the cursor is not updated unless I click on a button blindly. Is this a Wine specific problem?

**Comment by OmniBlade** [comment]
> When testing this with MinGW branch on Wine the cursor is not updated unless I click on a button blindly. Is this a Wine specific problem?

Could be wine specific or perhaps it needs bits from the other PRs I pushed to work correctly as I only tested it on top of the clean up stuff.

This might be structure packing now that I think about it, needs C flags as well as C++ flags which I had issues with in msvc as well.

**Review by hifi — APPROVED** [review]

**Pull request merged**